lib/ansible/vars/reserved.py

```diff
@@ -70,9 +70,11 @@ def get_reserved_names(include_private=True):
 
 def warn_if_reserved(myvars):
     ''' this function warns if any variable passed conflicts with internally reserved names '''
-    reserved = get_reserved_names()
-    for varname in myvars:
-        if varname == 'vars':
-            continue  # we add this one internally
-        if varname in reserved:
-            display.warning('Found variable using reserved name: %s' % varname)
+
+    varnames = set(myvars)
+    varnames.discard('vars')  # we add this one internally, so safe to ignore
+    for varname in varnames.intersection(_RESERVED_NAMES):
+        display.warning('Found variable using reserved name: %s' % varname)
```

Comment:
maybe just one warning line?

``` python
varname_conflicts = varnames.intersection(_RESERVED_NAMES)
if varname_conflicts:
    display.warning('Found variables using reserved names: %s' % ','.join(sorted(varname_conflicts)))
```

Or is the one warning per var so that the warning duplicate detector catches them? 
If it is for warning de-duping, should each play get a fresh set of warnings or just one globally for each varname?
(ie, include the play info in the warning msg...)